translator/batcher.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translator
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+	jaegerpb "github.com/jaegertracing/jaeger/model"
+)
+
+type bucketID [32]byte
+
+// spanBatcher is simpler version of OpenTelemetry's Node Batcher.
+// spanBatcher takes spans and groups them into Jaeger Batches using
+// the Span Process objects.
+type spanBatcher struct {
+	buckets map[bucketID]*jaegerpb.Batch
+}
+
+func (b *spanBatcher) add(span *jaegerpb.Span) {
+	if b.buckets == nil {
+		b.buckets = make(map[bucketID]*jaegerpb.Batch)
```

Comment:
nit: corner case: can this be called concurrently? if yes, it is possible that in theory a few spans are lost on the very first concurrent calls.

Reply:
It is not designed to be called concurrently and is private. The translator creates one batcher per list of spans it translates and adds them serially.